### PR TITLE
refactor map functions buildfilter and mergefilter

### PR DESCRIPTION
**Pull request by ciremusyoka**
Updated functions to match latest ones duplicated on gisida -react. 
Addressing https://github.com/onaio/gisida-react/issues/134

### CODE REVIEW

File: src/connectors/ona-api/auth.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export const ONAoauth = (reqConfig, token, dispatch) => api(reqConfig).then(({ u
     dispatch(receiveLogin(user));
     history.replace('/');
   }
-}).catch(err => console.log('Error: ', err));
+}).catch(err => console.log('Error: ', err));    // eslint-disable-line
```
Comment (cKellyDesign) [inline src/connectors/ona-api/auth.js]: Which rule is being violated here?

File: src/map/buildFilters.js [hunk 2]
```diff
@@ -29,52 +29,70 @@ export default function buildFilters(filters, layerFilters, prevFilters) {
...
       toggleAllOn: prevFilters
-        ? prevFilters[filterKey].toggleAllOn : true, // controls toggle all functionality and text
+        ? prevFilters[filterKey].toggleAllOn
+        : true, // controls toggle all functionality and text
       isFiltered: prevFilters
-        ? prevFilters[filterKey].isFiltered : false, // whether any options have been modified
+        ? prevFilters[filterKey].isFiltered
+        : false, // whether any options have been modified
       isOriginal: true, // whether the filter has been filtered
+      // eslint-disable-next-line
```
Comment (cKellyDesign) [inline src/map/buildFilters.js]: Which rule?

File: src/map/buildFilters.js [hunk 3]
```diff
@@ -29,52 +29,70 @@ export default function buildFilters(filters, layerFilters, prevFilters) {
...
           if (layerFilters[f][o] instanceof Array) {
-            const [first, second] = layerFilters[f][o];
-            filterKey = first;
-            optionKey = second;
-            filterMap[filterKey].options[optionKey].enabled = true;
-            filterMap[filterKey].options[optionKey].hidden = false;
-            if (!filterMap[filterKey]) filterMap[filterKey].isFiltered = true;
+            if (layerFilters[f][o][0] === '==') {
+              filterKey = layerFilters[f][o][1];  // eslint-disable-line
+              optionKey = layerFilters[f][o][2];  // eslint-disable-line
```
Comment (cKellyDesign) [inline src/map/buildFilters.js]: Which rule?

File: src/map/buildFilters.js [hunk 4]
```diff
@@ -29,52 +29,70 @@ export default function buildFilters(filters, layerFilters, prevFilters) {
...
+            }
           }
         }
       }
     }
   }
 
   return filterMap;
-}
+}
```
Comment (cKellyDesign) [inline src/map/buildFilters.js]: Let's not lose the trailing empty line pls

**Comment by ciremusyoka** [comment]
@cKellyDesign - Requested changes pushed.

**Comment by cKellyDesign** [comment]
@ciremusyoka Could you update this from staging and merge?

**Pull request merged**